dvc/repo/fetch.py

```diff
@@ -157,6 +157,10 @@ def fetch(  # noqa: PLR0913
     data, unversioned_count = _log_unversioned(data)
     failed_count += unversioned_count
 
+    if onerror:
+        for fs_index in data:
+            fs_index.onerror = _make_index_onerror(onerror, None)
```

Comment:
Hm, this normally should not be needed here, as fs_index should be already loaded. Need to take a closer look.